local-test/audit-dashboards.py

```
@@ -0,0 +1,256 @@
+def check_template_vars(dashboard: dict) -> list[str]:
+    declared = {v["name"] for v in dashboard.get("templating", {}).get("list", [])}
+    # Find every ${name} / $name reference in stringified dashboard
+    raw = json.dumps(dashboard)
+    used = set(re.findall(r"\$\{?(\w+)\}?", raw))
+    # Known Grafana built-ins we can skip
+    builtins = {
+        "__interval", "__interval_ms", "__range", "__range_s",
+        "__range_ms", "__rate_interval", "__timeFilter",
+        "__auto_interval", "__from", "__to", "__name",
+        "__user", "__org", "__dashboard", "__user_id",
+        "__auto", "__name", "__data",
```

Ruff (0.15.10)

[warning] Sets should not contain duplicate item `"__name"`

Remove duplicate item

(B033)